### PR TITLE
Rustfmt clippy

### PR DESCRIPTION
**Pull request by Manishearth**
Will review code changes myself.

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -316,25 +365,42 @@ pub fn is_negative(ty: LitIntType) -> bool {
...
-            Some(SignedIntLit(lty, s)) } else { None },
-        (UnsignedIntLit(lty), UnsignedIntLit(rty)) =>
+        (SignedIntLit(lty, _), SignedIntLit(rty, _)) => {
+            if lty == rty {
+                Some(SignedIntLit(lty, s))
+            } else {
+                None
+            }
+        }
+        (UnsignedIntLit(lty), UnsignedIntLit(rty)) => {
```
Comment (Manishearth) [inline src/consts.rs]: (reviewed till here)

File: src/methods.rs [hunk 2]
```diff
@@ -447,46 +467,162 @@ fn has_debug_impl<'a, 'b>(ty: ty::Ty<'a>, cx: &LateContext<'b, 'a>) -> bool {
...
-    ("from_iter",  1, NoSelf,     AnyType,  "std::iter::FromIterator"),
-    ("from_str",   1, NoSelf,     AnyType,  "std::str::FromStr"),
-];
+const CONVENTIONS: [(&'static str, &'static [SelfKind]); 5] = [("into_", &[ValueSelf]),
+                                                               ("to_", &[RefSelf]),
+                                                               ("as_", &[RefSelf, RefMutSelf]),
+                                                               ("is_", &[RefSelf, NoSelf]),
+                                                               ("from_", &[NoSelf])];
+
+const TRAIT_METHODS: [(&'static str, usize, SelfKind, OutType, &'static str); 30] = [("add",
```
Comment (sanxiyn) [inline src/methods.rs]: This looks especially bad.

File: src/approx_const.rs [hunk 3]
```diff
@@ -21,24 +21,22 @@ declare_lint! {
...
-    (f64::FRAC_PI_6, "FRAC_PI_6", 5),
-    (f64::FRAC_PI_8, "FRAC_PI_8", 5),
-    (f64::LN_10, "LN_10", 5),
-    (f64::LN_2, "LN_2", 5),
-    (f64::LOG10_E, "LOG10_E", 5),
-    (f64::LOG2_E, "LOG2_E", 5),
-    (f64::PI, "PI", 3),
-    (f64::SQRT_2, "SQRT_2", 5),
-];
+const KNOWN_CONSTS: &'static [(f64, &'static str, usize)] = &[(f64::E, "E", 4),
```
Comment (birkenfeld) [inline src/approx_const.rs]: This kind of change looks strange to me. It leaves tons of space on the left, and moves the important part quite a lot to the right.

Is there a rustfmt option to keep the original here?

Comment (Manishearth) [inline src/approx_const.rs]: I can't see one, but I could be wrong. Mostly okay with this formatting though.

**Comment by Manishearth** [comment]
Done reviewing. @llogiq @birkenfeld do you also want to go through this? Otherwise ready to merge.

The few cases of the const stuff -- we can look into that as a followup (since it's a few cases only), either with a rustfmt_skip or by tweaking rustfmt itself.

**Comment by llogiq** [comment]
I'll have a look later this evening (in 3-4 hours).

**Comment by llogiq** [comment]
I'm also not too fond of the "Visual" default formatting for many items; the code becomes quite right-biased this way. But I'm OK with merging; we can experiment with different settings later.

**Comment by Manishearth** [comment]
Yeah, I feel the same.

**Pull request merged**